src/components/nav/ConnectButton.tsx

```diff
@@ -24,6 +25,11 @@ export function ConnectButton() {
   const { openConnectModal } = useConnectModal()
   const { disconnect } = useDisconnect()
 
+  const onClickConnect = () => {
+    cleanupStaleWalletSessions()
+    openConnectModal?.()
```

Comment:
Why do you need an optional chaining here? as far as I see maybe there is no need for
![image](https://github.com/user-attachments/assets/60b99ec1-85fd-4ae2-a6e8-72de2aca6812)

Reply:
It's needed to make Typescript happy because openConnectModal can be undefined.
![image](https://github.com/user-attachments/assets/ca68ccd6-2400-48f9-b3d2-20ecb9050702)